### PR TITLE
Revert "Make HardwareConfig a record"

### PR DESCRIPTION
**Pull request by Gold856**
This reverts commit 1fb02a477d8e6106628d4a768fe3843b71ad217f.

## Description

Jackson constructs records with null/0 if the JSON field isn't present. We want it to be our defaults in code (empty String, empty List, etc). Revert back to the class structure to do this. Also adds tests to ensure defaults aren't null.

## Meta

Merge checklist:
- [x] Pull Request title is [short, imperative summary](https://cbea.ms/git-commit/) of proposed changes
- [x] The description documents the _what_ and _why_
- [ ] If this PR changes behavior or adds a feature, user documentation is updated
- [ ] If this PR touches photon-serde, all messages have been regenerated and hashes have not changed unexpectedly
- [ ] If this PR touches configuration, this is backwards compatible with settings back to v2025.3.2
- [ ] If this PR touches pipeline settings or anything related to data exchange, the frontend typing is updated
- [x] If this PR addresses a bug, a regression test for it is added

### CODE REVIEW

File: photon-core/src/main/java/org/photonvision/common/configuration/HardwareConfig.java [hunk 1]
```diff
@@ -21,57 +21,102 @@
 import java.util.ArrayList;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
-public record HardwareConfig(
-        String deviceName,
-        String deviceLogoPath,
-        String supportURL,
-        // LED control
+public class HardwareConfig {
```
Comment (samfreund) [inline photon-core/src/main/java/org/photonvision/common/configuration/HardwareConfig.java:24]: Can we add a comment here-ish about why we're reverting this so we don't accidentally make it a record again in the future?

**Review by samfreund — CHANGES_REQUESTED** [review]

File: photon-core/src/main/java/org/photonvision/common/configuration/HardwareConfig.java (hunk 1 above)
Comment (Gold856) [reply photon-core/src/main/java/org/photonvision/common/configuration/HardwareConfig.java:24]: Enforced by unit test

Comment (samfreund) [reply photon-core/src/main/java/org/photonvision/common/configuration/HardwareConfig.java:24]: I want the rationale documented.

Comment (spacey-sooty) [reply photon-core/src/main/java/org/photonvision/common/configuration/HardwareConfig.java:24]: Why? I'm ok with this changing so long as the test doesn't break

**Review by mcm001 — APPROVED** [review]
